### PR TITLE
fix(meta): check whether input sstable of compact task exist.

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

See https://github.com/risingwavelabs/risingwave/issues/10538.
Although https://github.com/risingwavelabs/risingwave/pull/10242 tried to fix this problem, but there is still some case occurs in compaction-group-split.

For example:
1. group-2 trigger a compact task with sstable-1 in sub level 1 and sstable-2 in sub level 2. The target sub level is 1.
2. group-2 split sstable-2 to new group-5. And we will change the origin `sst_id` of sstable-2 to `5`.
3. group-2 trigger a space-reclaim job to delete all files of which state-tables does not stay in current group. And it remove sstable-5 (the origin sstable-2).
4. The compact task in step-1 finished and it try to report to hummock-manager. But the origin sstable-2 has been removed in step-3. And `HummockVersion` would remove the sub level 2 since it had been an empty sub level. So the compact task can not find the sub level which it hope to insert new sstable files to.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/hummock/manager/mod.rs [hunk 1]
```diff
@@ -1273,10 +1283,20 @@ where
                     compact_task.set_task_status(TaskStatus::InputOutdatedCanceled);
                     false
                 } else {
-                    assert!(current_version
+                    let group = current_version
                         .levels
-                        .contains_key(&compact_task.compaction_group_id));
-                    true
+                        .get(&compact_task.compaction_group_id)
+                        .unwrap();
```
Comment (zwang28) [inline src/meta/src/hummock/manager/mod.rs:1289]: can be none

Comment (Little-Wallace) [reply src/meta/src/hummock/manager/mod.rs:1289]: No. It must exist. Because we have check whether it exist before.

**Review by zwang28 — APPROVED** [review]
LGTM

**Review by hzxa21 — APPROVED** [review]

File: src/meta/src/hummock/manager/compaction_group_manager.rs [hunk 2]
```diff
@@ -630,6 +635,34 @@ impl<S: MetaStore> HummockManager<S> {
         new_version_delta.commit();
         branched_ssts.commit_memory();
         self.notify_last_version_delta(versioning);
+        drop(versioning_guard);
+        let mut canceled_tasks = vec![];
```
Comment (Li0k) [inline src/meta/src/hummock/manager/compaction_group_manager.rs:639]: we can add example or issue link for comment this case

**Review by Li0k — APPROVED** [review]
Rest LGTM

**Pull request merged**